src/calculators/BarnesPointsCalc.ts

```diff
@@ -0,0 +1,134 @@
+import { Results } from "../common/CrewTimerTypes";
+import {
+  genPlaces,
+  isAFinal,
+} from "../common/CrewTimerUtils";
+
+export type BarnesPointsResults = {
+  team: string;
+  points: number;
+  place: number;
+}[];
+
+export const maxPointsFromName = (eventName: string) => {
+  eventName = eventName.toUpperCase();
+  const boatClassCaptureExpression = /.* ([1248]+)[x\-\+]+/;
```

Comment:
I think you don't need the + after each[] pattern as you don't want to allow more than digit or suffix.